lib/Data/Time/Format/Parse.hs

```diff
@@ -92,6 +92,20 @@ class ParseTime t where
 --
 -- [@%0f@] accepts exactly two digits.
 --
+-- For example, to parse a date in YYYY-MM-DD format, while allowing the month 
+-- and date to have optional leading zeros (notice the @-@ modifier used for @%m@
+-- and @%d@:
+--
+-- > GHCi> parseTimeM True defaultTimeLocale "%Y-%-m-%-d" "2010-3-04" :: Maybe Day
+-- > Just 2010-3-04
+-- 
+-- __Caution:__ While you can use any Monad for 'parseTimeM', please be cautioned 
```

Comment:
Omit this paragraph.

Reply:
Can we let this remain? I was bitten by this issue  because I was expecting the Left value to be returned. My guess is that a lot of newbies wont realise whats going on till a parsing fails and blows up with a  runtime error!

Reply:
I'd like to keep the documentation concise, and focus only on issues specific to the library. `Either String a` is never a way of catching `fail` exceptions.

Reply:
Can we at least keep a note about the fact that parseTimeM uses fail and may result in runtime errors depending upon the monad that one may use?

Reply:
So, the problem is that `parseTimeM` and `parseTimeError` don't behave differently if you end-up using the incorrect monad. That's the only thing I'd like to point out in the docs. Having something blow-up with a runtime error in Haskell is a pretty bad dev-UX IMO. Final call is your. I'd propose *at least* having the following in the `parseTimeM` docs:

> **Caution:** `parseTimeM` use `fail`in-case an input string could not be parsed; depending upon the monad that you use, this _may_ result in an error being thrown.

Reply:
Hmm, I think the correct thing to do is to change the type of `parseTimeM` to require `MonadFail` (https://github.com/haskell/time/issues/79). Once that's fixed, it won't be possible to use `Either`.

Reply:
I do intend to do a 1.8.0.4 release that includes this PR, but not #79. However, I'd rather not add the text if I'm just going to remove it again for 1.9.

Reply:
I've removed the cautionary paragraph. You may want to consider leaving the paragraph there for the 1.8.x branch for people who are not upgrading to the 1.9 branch.